addons/craft/create_recipes.py

```diff
@@ -80,13 +111,16 @@ def get_recipes_from_soup(soup, spheres=False):
             row
             for row in subtable.children
             if (hasattr(row, 'findAll') and
-                len(row.findAll('td')) == count)
+                len(row.findAll('td')) in lengths)
         ]
         rows.extend(children)
     return get_recipes_from_rows(rows, spheres)
 
 def get_items_dictionary():
-    path = 'C:\\Program Files (x86)\\Windower4\\res\\items.lua'
+    if platform.system() == 'Windows':
+        path = 'C:\\Program Files (x86)\\Windower4\\res\\items.lua'
+    else:
+        path = os.path.join(os.path.expanduser("~"), 'Resources/lua/items.lua')
```

Comment:
Is the working directory this is exceuted in the addon directory? If so, you can work backwards from here and just go to `..\..\res\items.lua`. Then you need no distinction by OS, nor an assumption about the installation path.

Reply:
There are two ways to run the script.

1 - The way you've suggested, which is more of a FFXI user way.  If there are new recipes posted on bg-wiki and people don't want to wait for an update, they can run the script themselves from the addon directory.

2 - Running the script from a forked repository.  This is more of the developer way and the way I usually do it.  On windows, I have git repository cloned in C:/Lua and just use the windower resource files.  This lets me change the addon source code without having to worry about the launcher overwriting (or not overwriting) whatever I'm working on.  On Linux I don't have the client or windower installed, I rely on ~/Lua to run the script and ~/Resources for the resources (both cloned repositories.)

Reply:
The overall point being is that if I wish to be able to run the script out of a cloned Lua repository, I can't use a relative path because the resources are not included with the Lua repository.